pandas/core/frame.py

```diff
@@ -451,6 +461,89 @@
 1   foo      8
 2   bar      7
 3   bar      8
+
+Merge dataframes `left` and `right`, where column `left.timestep` falls
+within the range of `right.timestart` to `right.timeend`, using `on` as a
+callable.
+
+>>> left = pd.DataFrame({'timestep': range(5)})
+>>> right = pd.DataFrame({'mood': ['happy', 'jolly', 'joy', 'cloud9'],
+...                       'timestart': [0, 2, 2, 3],
+...                       'timeend': [1, 3, 4, 4]})
+>>> left
+   timestep
+0         0
+1         1
+2         2
+3         3
+4         4
+
+>>> right
+     mood  timestart  timeend
+0   happy          0        1
+1   jolly          2        3
+2     joy          2        4
+3  cloud9          3        4
+
+>>> merge = left.merge(
+...     right,
+...     on=lambda left, right: (right.timestart <= left.timestep)
+...                            & (left.timestep <= right.timeend)
+... )
+
+>>> merge
+    timestep    mood  timestart  timeend
+0          0   happy          0        1
+1          1   happy          0        1
+2          2   jolly          2        3
+3          2     joy          2        4
+4          3   jolly          2        3
+5          3     joy          2        4
+6          3  cloud9          3        4
+7          4     joy          2        4
+8          4  cloud9          3        4
+
+Conditional merge on with an equality and range condition
+
+>>> left = pd.DataFrame(
+...     {
+...         'fdate': pd.DatetimeIndex(['1982-06-30'] * 5),
```

Comment:
Could you give these example better columns names like `"date"`, `"value"`, etc.

Reply:
done